awx/main/models/workflow.py

```diff
@@ -472,6 +480,16 @@ def notification_data(self):
     def task_impact(self):
         return 0
 
+    def get_ancestor_workflows(self):
+        ancestors = []
+        wj = self
+        while True:
+            wj = wj.get_workflow_job()
+            if (not wj) or (not wj.workflow_job_template_id):
+                break
+            ancestors.append(wj.workflow_job_template_id)
+        return ancestors
```

Comment:
might want to track a set of workflow jobs, so that we can bail if we find that this loop is repeating itself. The tracking of the templates will not provide that, and right now it is vulnerable to an infinite loop if someone created cycles via the shell.